src/app/my-rewards/backers/components/RBI.tsx

```diff
@@ -100,5 +101,18 @@ export const RBI: FC<RBIProps> = props => {
   const {
     flags: { use_state_sync },
   } = useFeatureFlags()
-  return use_state_sync ? <RBIWStateSync {...props} /> : <RBIWTheGraph {...props} />
+
+  const {
+    data: stateSyncIsHealthy,
+    isLoading: healthCheckIsLoading,
+    error: healthCheckError,
+  } = useStateSyncHealthCheck()
+
+  useHandleErrors({ error: healthCheckError, title: 'State sync health check error' })
+
+  return use_state_sync && !healthCheckIsLoading && stateSyncIsHealthy ? (
```

Comment:
I tried it and when the state sync is healthy it always doing an attempt in the graph and then it start querying form the state sync, which should avoid that since the graph have a cost per request and should be only the fallback.

Reply:
I'm not sure if I'm interpreting your message right, but if I do, then I cannot reproduce it.
Forcing the health check to return true seems to always result in calling the state sync only.